result/capture.yaml

```diff
@@ -9,34 +9,35 @@ components:
   schemas:
     Result.CaptureRequest:
       description: >-
-        The capture result request to the traffic generator 
+        The capture result request to the traffic generator.
+        Stops the port capture.
       type: object
+      required: [port_name]
       properties:
-        port_names:
+        port_name:
```

Comment:
is there a reason now this only stops a single port instead of list of ports?  same question as above for Control.PortCapture..

Reply:
I envision taking a capture as two steps - much like you do when you use tcpdump, start capturing (provide all details such as filters etc), then stop capturing (return the capture).  
No saved files, no special file locations, no special filenames (special characters, linux/windows), no download links, download format (multipart, octect-stream) etc.

Reply:
i guess the trouble i have reconciling is what is the expected output to the user if say, they started a capture at time x for port abc, and say issue a results for port abc's capture at time x+ 1 hour later.  I'm just wondering as a user, what I'd expect to be in that response for the results....just seems we can always run into some kind of resource limits on the TG side of things pending how much is captured.

Reply:
Resource limits would be implementation specific. Since the proposed schema of capture start is an object it can be extended with additional properties as needed (can add a capture limit, stop after x seconds etc) rather than trying to figure out all of those details now.  The current modeled API is start capture (on x number of ports with optional filters). Get a port's capture when the need arises. Implementations can cap however they want and expose that in the /results/capabilities API.

The sample illustrates starting a capture on a port, getting the capture and writing the capture to a file.  This leaves the file name and the file location up to the user. 
```
api.set_port_capture({'port_names': ['p1'], 'filters': null, 'format': 'pcap'})
capture = api.get_capture_results({'port_name': 'p1'})
with open('p1_%s.cap' % time.time(), 'wb') as fid:
    fid.write(capture)
```

Reply:
i guess what i meant is there will be implementation specific cap and the question is how would a user know whether in the example code above whether the received data spans the first 5 minutes or the last 5 minutes of the capture, assuming time between start capture and results is say 1 hour, and the target only has a 5 minute worth of buffer (assuming some rate)?  is there a way to maybe signal some metadata back assuming we hit some kind of cap in the returned result?  Sorry, i'm asking all these w/o a real use case in mind....maybe overthinking it.

Reply:
Agree.  I have some of these same questions in mind and it is not clear to me which way things should go.  My initial reaction was that I think of stopping and getting the capture as two distinct steps and I didn't like the fact that start is an action under 'control'  but stop is an action under 'results'.  When I think about it further, it isn't that obvious to me.   And I still have the same questions about limits and stuff how things would behave with those constraints.  Need some solid use-cases in mind.

Reply:
i thought pcap has timestamps?

Time stamp usage considerations
Now, there are several important aspects that every network forensic specialist needs to be aware of. Firstly, timestamps are derived from the clock on the machine performing the packet capture. Therefore it’s important to check the accuracy and record this machine’s system time before starting the packet capture.

Secondly, time stamps are saved to the .pcap file in GMT/UTC format, which can easily be verified by examining the value of “thiszone” in the global header. For display purposes, packet capture software such as Wireshark automatically converts these timestamps into the local time of the machine used to examine the .pcap files. However UTC times can also be displayed in Wireshark by adjusting its “Column Preferences”.

Reply:
I'm trying to avoid having information in the response that is duplicated - such as timestamps which is already included in the pcap.
Limits (and any other request information) can be added to /control/capture request since its an object - i just don't want to add it right now since there is no specific use case for it.

Reply:
I'd assume the RDMA use case right now is setting up traffic in a way where it is probably a fixed amount that one would know it will never exceed any practical resource limit on the device performing the capture?  So their use case should be okay.  I guess my question is do we want to get the implementation to return a signal to the end user indicating whether the capture may be truncated or not (due to resource limit, whether a capacity limit in the capture buffer, or it has wrapped a circular buffer), just a simple true/false..  This is so that if it comes back as "true", the user at least have a definitive hint on whether he needs to put in better capture filters, or set up the traffic in a way where he won't exhaust the resource limit on capture.

Reply:
The OpenAPI standard allows for streaming and non streaming responses.
A non-streaming response will be available for those implementations that can only return a capture file when requested.
A streaming response will be available for those implementations that do not want to incur storage overhead and put the onus on the user to interpret/store/stop the stream of capture results.

Reply:
just occurred to me, for streaming response, how does the user stop the capture?